certification/internal/shell/imagesource_registry.go

```diff
@@ -0,0 +1,78 @@
+package shell
+
+import (
+	"bytes"
+	"strings"
+
+	cmdchain "github.com/rainu/go-command-chain"
+	"github.com/redhat-openshift-ecosystem/openshift-preflight/certification"
+	log "github.com/sirupsen/logrus"
+)
+
+type ImageSourceRegistryCheck struct {
+}
+
+var approvedRegistries = map[string]string{
+	"registry.connect.dev.redhat.com":   "registry.connect.dev.redhat.com",
+	"registry.connect.qa.redhat.com":    "registry.connect.qa.redhat.com",
+	"registry.connect.stage.redhat.com": "registry.connect.stage.redhat.com",
+	"registry.connect.redhat.com":       "registry.connect.redhat.com",
+	"registry.redhat.io":                "registry.redhat.io",
+	"registry.access.redhat.com":        "registry.access.redhat.com",
+}
+
+func (p *ImageSourceRegistryCheck) Validate(bundleImage string) (bool, error) {
+
+	output := &bytes.Buffer{}
+
+	err := cmdchain.Builder().
+		WithInput(strings.NewReader(bundleImage)).
+		Join("cut", "-d", ",", "-f1").
+		Join("cut", "-d", "/", "-f1").
+		Finalize().WithOutput(output).Run()
+	if err != nil {
+		log.Error(" Failed to execute cmdchain builder")
+		log.Debug(" failed to execute cmdchain builder", err)
+		return false, nil
+	}
+
+	userRegistry := strings.TrimRight(output.String(), "\n")
```

Comment:
Is there whitespace to the "left" of the string that is desirable? Otherwise, it's better to just use TrimSpace. You don't have to specify potentially platform dependent values.

Reply:
Thanks. Code Improved.